pages/api/web-scraping/flights.js

```diff
@@ -0,0 +1,187 @@
+// @ts-check
+import { FingerprintJsServerApiClient, Region } from '@fingerprintjs/fingerprintjs-pro-server-api';
+import { CheckResult, checkResultType } from '../../../server/checkResult';
+import { isRequestIdFormatValid, originIsAllowed, visitIpMatchesRequestIp } from '../../../server/checks';
+import { ALLOWED_REQUEST_TIMESTAMP_DIFF_MS, SERVER_API_KEY } from '../../../server/const';
+import { sendErrorResponse, sendForbiddenResponse, sendOkResponse } from '../../../server/response';
+import { ensureGetRequest, messageSeverity } from '../../../server/server';
+import { DAY_MS, FIVE_MINUTES_MS, HOUR_MS } from '../../../shared/const';
+import { AIRPORTS } from '../../web-scraping';
+
+/**
+ * @typedef {Object} ResultsQuery
+ * @property {string} from
+ * @property {string} to
+ * @property {string} requestId
+ * */
+
+/**
+ * @param {import('next').NextApiRequest} req
+ * @param {import('next').NextApiResponse} res
+ */
+export default async function getFlights(req, res) {
+  // This API route accepts only GET requests.
+  if (!ensureGetRequest(req, res)) {
+    return;
+  }
+
+  const { from, to, requestId } = /** @type {ResultsQuery} */ (req.query);
+
+  // Validate request ID format
+  if (!isRequestIdFormatValid(requestId)) {
+    sendForbiddenResponse(
+      res,
+      new CheckResult('Invalid request ID.', messageSeverity.Error, checkResultType.RequestIdMismatch)
+    );
+    return;
+  }
+
+  try {
+    // Retrieve analysis event from the Server API using the request ID
+    const client = new FingerprintJsServerApiClient({ region: Region.Global, apiKey: SERVER_API_KEY });
+    // If the requestId does not exist, the SDK will throw an error which will be caught below
+    const eventResponse = await client.getEvent(requestId);
+    const botData = eventResponse.products?.botd?.data;
+
+    if (!botData) {
+      sendOkResponse(
+        res,
+        new CheckResult(
+          'Bot detection is disabled, access allowed.',
+          messageSeverity.Success,
+          checkResultType.Passed,
+          getFlightResults(from, to)
+        )
+      );
+      return;
+    }
+
+    if (botData.bot?.result === 'good') {
+      sendOkResponse(
+        res,
+        new CheckResult(
+          'Access allowed, good bot detected.',
+          messageSeverity.Success,
+          checkResultType.GoodBotDetected,
+          getFlightResults(from, to)
+        )
+      );
+      return;
+    }
+
+    // Check for bot presence and type
+    if (botData.bot?.result === 'bad') {
+      sendForbiddenResponse(
+        res,
+        new CheckResult(
+          '🤖 Malicious bot detected, access denied.',
+          messageSeverity.Error,
+          checkResultType.MaliciousBotDetected
+        )
+      );
+      // Optionally, here you could also save the bot's IP address to a blocklist in your database
+      // and block all requests from this IP address in the future at a web server/firewall level.
+      return;
+    }
```

Comment:
Maybe we could extract these botd-related checks into separate functions that could be resued in future use cases?

Reply:
Do you mean the condition `botData.bot?.result === 'bad'` ? Or the entire block? 

If you guys prefer to separate it I will separate it :) But I have written it this way deliberately. Here's why: 

This is partially subjective/convention, so I am happy to respect your original choices but based on my experience of exploring the codebase from a fresh newcomer's perspective, I think it would make the codebase more approachable if we try to limit the number of files and abstractions per use case, even at the cost of slightly larger files, longer functions or some repetition. I want to minimize users having to "Go to definition" several times, and getting lost switching between multiple files to understand how the use case works. 

Right now, you can open two files (`web-scraping/index.jsx` and `flights.js`) and understand the essence of the entire implementation. There is a page and an endpoint the page calls and *everything* you need to know is in these two files --  plus some simple functions and presentation components, but always *just one level deep*. From a teaching/learning standpoint, I think that's valuable and we should have a good reason to trade it away. 
Every file and level of abstraction/function wrapper makes the example less readable (from my POV at least).  

Obviously, if a file gets too large or a function too complex, you need to break it up, but I don't think this one is there yet. There is always a compromise between readability and DRYness. In an open-source learning resource, I would favor readability. Especially if don't have another Bot detection use case, there is no upside to creating the abstraction *yet*. 

So we can 
A) revisit this when we get a second Bot detection use case and have a better idea of what shape of abstraction we need
B) separate it now, it's not a big deal (I overthink these things sometimes and I could be wrong, if so, tell me how :) 

Let me know what you think :)

Reply:
Makes sense! Let's go with A) approach for now :)